rust-runtime/aws-smithy-runtime-api/src/client/runtime_components.rs

```diff
@@ -612,11 +612,23 @@ impl RuntimeComponentsBuilder {
         &mut self,
         scheme_id: AuthSchemeId,
         identity_resolver: impl ResolveIdentity + 'static,
+    ) -> &mut Self {
+        self.set_shared_identity_resolver(scheme_id, identity_resolver.into_shared())
+    }
+
+    /// Sets the identity resolver for a given `scheme_id`.
+    ///
+    /// This is effectively the same as [`Self::set_identity_resolver`] but takes a [`SharedIdentityResolver`]
+    /// instead.
+    pub fn set_shared_identity_resolver(
```

Comment:
Can we use the `IntoShared` trait to modify the existing `set_identity_resolver` function in a backwards compatible manner?

Reply:
Thanks for pointing it out! Turns out that this method is unnecessary because `set_identity_resolver` can already take a `SharedIdentityResolver` and `.into_shared()` is smart enough to not wrap a `SharedIdentityResolver` with another `SharedIdentityResolver` (as tested [here](https://github.com/smithy-lang/smithy-rs/blob/8873666b0ee0cf6dada9599778c96c77580f9042/rust-runtime/aws-smithy-runtime-api/src/shared.rs#L227-L231)).

Fixed in ef1e1be.